### PR TITLE
#75 Badge scan should allow to switch between sponsor and attendee

### LINKED ISSUE #75: Sponsor badge scanning improvements
* CHANGE: sponsors do NOT have to login using a social network. they select: I'm a sponsor -> select their sponsor -> type the (simple) password -> scan -> badges are synced to CloudLink in 1 list per sponsor
* when making a attendee / sponsor selection, there should be a way to go back
* when already made a choice between attendee / sponsor, there should be a way to change without the need to logout

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Selecting an option from MyBadge view now allows to switch selection between Attendee and Sponsor. A user can also logout when signed in either as Sponsor or Attendee.

Fixes #75

### CODE REVIEW

File: DevoxxClientMobile/src/main/java/com/devoxx/service/DevoxxService.java [hunk 1]
```diff
@@ -765,9 +765,6 @@ private void retrieveExhibitionMapsInternal() {
 
     @Override
     public ObservableList<SponsorBadge> retrieveSponsorBadges() {
-        if (!isAuthenticated() && DevoxxSettings.USE_REMOTE_NOTES) {
-            throw new IllegalStateException("An authenticated user must be available when calling this method.");
-        }
         
         if (sponsorBadges == null) {
             final GluonObservableList<SponsorBadge> sponsorBadges = internalRetrieveSponsorBadges();
```
Comment (jperedadnr) [inline DevoxxClientMobile/src/main/java/com/devoxx/service/DevoxxService.java]: There is a NPE here when you try to sign in as a sponsor without having an authenticated user. 
When calling `internalRetrieveSponsorBadges`, if `authenticationClient.getAuthenticatedUser()` is null, `authenticationClient.getAuthenticatedUser().getKey()` throws the NPE.

File: DevoxxClientMobile/src/main/java/com/devoxx/DevoxxApplication.java [hunk 2]
```diff
@@ -221,6 +219,17 @@ public Button getSearchButton() {
         return navSearchButton;
     }
     
+    public MenuItem scanAsDifferentUser() {
+        final MenuItem scanAsDifferentUser = new MenuItem(DevoxxBundle.getString("OTN.BADGES.SCAN.DIFFERENT.USER"));
```
Comment (jperedadnr) [inline DevoxxClientMobile/src/main/java/com/devoxx/DevoxxApplication.java]: After some sign in/logout attempts, the `Scan with Different User` button shows a wrong 
[menu item (click for image)](https://www.dropbox.com/s/ymq4df6pwzl6mwx/badges.png?dl=0).

Comment (abhinayagarwal) [reply DevoxxClientMobile/src/main/java/com/devoxx/DevoxxApplication.java]: This bug is random and extremely difficult to re-create. The location of the MenuPopupView is correct. However, the size of the ScrollPane restricts Label from expanding. My guess would be that it has to do something with how we resize rootNode in `PopupView#layoutChildren()` but its extremely difficult to analyze without reproduction of the bug.

File: DevoxxClientMobile/src/main/java/com/devoxx/views/BadgesPresenter.java [hunk 3]
```diff
@@ -84,100 +92,136 @@ public void initialize() {
...
-                    Badge badge = new Badge(qr);
-                    if (badge.getBadgeId() != null) {
-                        boolean exists = false;
-                        for (Badge b : badges) {
-                            if (b.getBadgeId().equals(badge.getBadgeId())) {
-                                Toast toast = new Toast(DevoxxBundle.getString("OTN.BADGES.QR.EXISTS"));
-                                toast.show();
-                                exists = true;
-                                break;
+        getApp().getAppBar().getMenuItems().setAll(getBadgeChangeMenuItem("Logout"));
```
Comment (jperedadnr) [inline DevoxxClientMobile/src/main/java/com/devoxx/views/BadgesPresenter.java]: Move to bundle

**Pull request merged**